pathToSpeaker simulating using limelight odometry and y button for control

grace-mills opened the pull request.

grace-mills committed:
Merge remote-tracking branch 'origin/main' into gmills

grace-mills committed:
Merge remote-tracking branch 'origin/main' into gmills

grace-mills committed:
Merge remote-tracking branch 'origin/main' into gmills

grace-mills committed:
adding vision to RobotContainer

grace-mills committed:
Odometry does not reset to zero when removing april tag

done in comp2023- imported using swerve subsystem

grace-mills committed:
checkIfNonZero method

grace-mills committed:
2024 simulation without going back to origin with no apriltag reading

grace-mills committed:
Merge remote-tracking branch 'origin/main' into gmills

grace-mills committed:
Merge remote-tracking branch 'origin/main' into gmills

grace-mills committed:
pathToSpeaker on y button

frc2135 committed:
Created pathToSpeaker path

frc2135 committed:
pathToSpeaker auto

grace-mills committed:
path to speaker works in simulation using y button

jmullins16 commented on Comp2023-Imported/src/main/java/frc/robot/subsystems/Swerve.java:
```diff
@@ -186,12 +186,21 @@ public void updateSwerveOdometry( )
+      Rotation2d OriginRot = Origin.getRotation( );
 
-      if (rawPose != null)
+      if (rawPose != null && rawPose.getX( ) != OriginX && rawPose.getY( ) != OriginY && rawPose.getRotation( ) != OriginRot)
```
Look at this API page at the "equals" method for Pose2d. Should be able to do something like:

if (rawPose != null || !rawPose.equals(origin))

https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/math/geometry/Pose2d.html#%3Cinit%3E()

jmullins16 commented on Comp2023-Imported/src/main/java/frc/robot/subsystems/Swerve.java:
```diff
@@ -186,12 +186,21 @@ public void updateSwerveOdometry( )
       if (botLLPose != null && DriverStation.isTeleopEnabled( ))
+      {
         m_poseEstimator.addVisionMeasurement(botLLPose, Timer.getFPGATimestamp( ) - (latency / 1000));
+        m_poseEstimator.update(m_heading, getPositions( ));
```
I think this .update call should already be called within the swerve library provided by CTRE (or else it would not have worked at all even when the LL wasn't there). So this might be applying the update a second time.

jmullins16 commented on Comp2023-Imported/src/main/java/frc/robot/subsystems/Swerve.java:
```diff
@@ -505,9 +514,12 @@ public Pose2d getPose( )
+    DataLogManager.log("heading!!!!!!!!!!!!: " + m_heading);
+    for (int i = 0; i < getPositions( ).length; i++)
+    {
+      DataLogManager.log("positions!!!!: " + i + " " + getPositions( )[i]);
```
The pose is useful, but I'm not sure logging the individual swerve module positions is helpful. We should discuss.

jmullins16 commented on Comp2024/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java:
```diff
@@ -30,11 +32,12 @@
   private static final double                    kSimLoopPeriod = 0.005; // 5 ms
   private Notifier                               m_simNotifier  = null;
   private double                                 m_lastSimTime;
+  private final boolean                          m_useLimelight = true; // set to false when no limelight to prevent sim errors
```
When checking in the code, we need to leave this as "false" to prevent everyone else from having simulation problems. When we get working robots, we can leave it as "true". You'll need to set it to true when you are testing.

jmullins16 commented on Comp2024/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java:
```diff
@@ -135,11 +138,18 @@ public void periodic( )
       Pose2d llPose = lastResult.getBotPose2d_wpiBlue( );
 
-      if (lastResult.valid)
+      if (lastResult.valid && llPose.getX( ) != 0 && llPose.getY( ) != 0)
```
Did we prove that we get a 0,0,0 pose with the valid setting as true? I don't remember the results.

grace-mills replied:
Yes valid is a boolean in Limelight Helpers and we still had the 0,0,0 problem with it true

jmullins16 commented on Comp2024/src/main/java/frc/robot/subsystems/Vision.java:
```diff
@@ -82,12 +82,19 @@ public void periodic( )
   public void simulationPeriodic( )
   {
-    // This method will be called once per scheduler run when in simulation
+    // Pose2d llPose = getLimelightRawPose( );
```
Maybe these lines aren't needed anymore? It's commented out.

jmullins16 reviewed (changes requested):
Looks pretty good. We should discuss my questions before making new changes and merging.

grace-mills committed:
Fixing PR comments

driveToSpeaker method commented out because not fixed to be a path generated on the fly yet

grace-mills committed:
Merge remote-tracking branch 'origin/main' into gmills

jmullins16 commented on Comp2023-Imported/src/main/java/frc/robot/subsystems/Swerve.java:
```diff
@@ -186,12 +186,16 @@ public void updateSwerveOdometry( )
+      }
 
       Pose2d rawPose = m_vision.getLimelightRawPose( );
+      Pose2d Origin = new Pose2d(new Translation2d(0.0, 0.0), new Rotation2d(0));
```
This is a minor "nit", but origin is a local variable and should be "origin" (drinking camel case).

jmullins16 replied:
I guess this is in Comp2023-imported anyway and not our Comp2024 codebase.

jmullins16 reviewed (approved):
Approving with only a minor request for capitalization of a variable. For the LL returning a zero pose, you're doing the right thing by keeping the check in the code for now and getting it merged. I believe that we'll learn a better way to detect the "valid" state in the future, because the LL has all of the information needed to mark a (0, 0, 0) pose as invalid--we've probably just not figured it out. But now is not the time to go down that rabbit hole. Nice work with all of the changes!
